src/components/MaterialUI/DataDisplay/Table.js

```diff
@@ -1,64 +1,361 @@
+	},
+	table: {
+		tableLayout: "fixed",
+	},
+	stickyHeaderHead: {
+		borderTop: props =>
+			props.withoutTopBorder ? "none" : "1px solid " + theme.palette.grey.borders,
+		borderBottom: "1px solid " + theme.palette.grey.borders,
+		backgroundColor: theme.palette.grey.lighter,
+		display: "flex",
+		overflow: "hidden",
+		cursor: "default",
+	},
+	stickyHeaderTable: {
+		tableLayout: "fixed",
+		width: props => "calc(100% - " + (props.scrolled || "0") + "px)",
+	},
+	stickyHeaderTableScroll: {
+		width: props => (props.scrolled || "0") + "px",
+		backgroundColor: theme.palette.grey.lighter,
+		display: props => (props.scrolled > 0 ? "block" : "none"),
+	},
+	tableHeader: {
+		borderTop: props =>
+			props.withoutTopBorder ? "none" : "1px solid " + theme.palette.grey.borders,
+		borderBottom: props =>
+			props.stickyHeader ? "none" : "1px solid " + theme.palette.grey.borders,
+		display: props => (props.stickyHeader ? "none" : "table-header-group"),
+		backgroundColor: theme.palette.grey.lighter,
+		overflow: "hidden",
+		cursor: "default",
+	},
+	tableBody: {
+		overflowX: "hidden",
+		overflowY: "auto",
+	},
+	tableRow: {
+		borderBottom: "1px solid " + theme.palette.grey.borders,
+		cursor: props => (props.onRowClick ? "pointer" : "default"),
+		backgroundColor: theme.palette.background.paper + "!important",
+		"&:hover": {
+			backgroundColor: props =>
+				props.onRowClick ? theme.palette.grey.lighter : theme.palette.background.paper,
+		},
+	},
+	tableCell: {
+		padding: "20px 16px",
+		whiteSpace: "nowrap",
+		overflow: "hidden",
+		textOverflow: "ellipsis",
+	},
+	tableCellSelect: {
+		padding: "17px 16px",
+		width: "50px",
+	},
+	headerCell: {
+		padding: "11px 16px",
+		textAlign: "left",
+		fontWeight: theme.typography.fontWeightBold,
+	},
+	headerCellSelect: {
+		padding: "8px 16px",
+		width: "50px",
+		textAlign: "left",
+		fontWeight: theme.typography.fontWeightBold,
+	},
+	rowSelectCheckbox: {
+		padding: 0,
 	},
 }));
 
-const Table = ({ headers, rows, checkbox, placeholder }) => {
-	const classes = useStyles();
+function propsAreEqualRow(prev, next) {
+	return prev.selected === next.selected;
+}
+
+export const MemoTableRow = React.memo(TableRow, propsAreEqualRow);
+
+function propsAreEqualBody(prev, next) {
+	return (
+		next.contentTimestamp !== undefined &&
+		prev.contentTimestamp === next.contentTimestamp &&
+		prev.selectedNumber === next.selectedNumber
+	);
+}
+
+const TableBody = props => {
+	return <tbody className={props.className}>{props.rows}</tbody>;
+};
+
+export const MemoTableBody = React.memo(TableBody, propsAreEqualBody);
+
+const TableCell = withDeferredTooltip(
+	React.forwardRef((props, ref) => {
+		return (
+			<td className={props.className} ref={ref} {...props}>
+				{props.value}
+			</td>
+		);
+	}),
+);
+
+// We will assume for now that cell will never change value so we want to avoid rerendering as much as possible
+export const MemoTableCell = React.memo(TableCell, () => true);
```

Comment:
Sounds like potential possible "bug" in future :) 
Is that really affects performance so much to have correct compare function right now?

Reply:
I thought about it, don't worry, problem is I cannot compare "value" as it may be component, which could be different everytime even though it might not be in fact. I would need primitive type to ensure it is not rerendered everytime. Will do some more tests.

Reply:
Yeah understand the problem with components.. Maybe it won't hurt to much if cell components will be rerenderred. Also, what I'm suggesting as "naive" compare scenario for components we can implement is to try check if all children and value properties inside that component are the same (most of the time values are inside those 2 properties). For that I would suggest to implement helper function which will retrieve values for all the "children" and "value" properties for passed component.

Reply:
Using TableCell directly is fine enough after some exhaustive testing. I will remove MemoTableCell as maybe it is some legacy code of an earlier design or previous testing I had performed.